Failed to build arm64 architecture related board

**Describe the bug**
I was trying to build the zephyr on qemu_cortex_a53, bcm958402m2_a72, xenvm using command eq:$west build -b qemu_cortex_a53 samples/hello_world and it failed with error attached in the log section. it seems like environment issue, but i reinstall zephyr followed the getting started guide but no luck.

I have tried 
1)	Switch back to the changes https://github.com/zephyrproject-rtos/zephyr/commit/3539c2fbb35f3e85f2b6348b577fa3b3378a97b6 Minus 1 => Able to build

2)	Try to find on internet, what environment variable might possibly causing the issue, and play around with setting the variable => failed with the same error
		a.	ZEPHYR_BASE
		b.	ZEPHYR_TOOLCHAIN_VARIANT
		c.	ZEPHYR_SDK_INSTALL_DIR
		d.	CMAKE_C_COMPILER

3)	Updated everything according to the getting started guide => failed with the same errror

4)	Reinstall zephyr and follow every steps in getting started guide on another machine => failed with the same error


**To Reproduce**
Steps to reproduce the behavior:
$west build -b qemu_cortex_a53 samples/hello_world

**Expected behavior**
Build pass.

**Impact**
Showstopper

**Logs and console output**
-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=/usr/bin/python3 -B/home/boonkhai/zephyrtesting/zephyrproject/zephyr/build -S/home/boonkhai/zephyrtesting/zephyrproject/zephyr/samples/hello_world -GNinja -DBOARD=qemu_cortex_a5
boonkhai@pg-centralnode-1:~/zephyrtesting/zephyrproject/zephyr$ west build -b qemu_cortex_a53 samples/hello_world                                              -- west build: generating a build system
Including boilerplate (Zephyr base (cached)): /home/boonkhai/zephyrtesting/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/boonkhai/zephyrtesting/zephyrproject/zephyr/samples/hello_world
-- Zephyr version: 2.5.99 (/home/boonkhai/zephyrtesting/zephyrproject/zephyr)
-- Found west (found suitable version "0.10.1", minimum required is "0.7.1")
-- Board: qemu_cortex_a53
-- Cache files will be written to: /home/boonkhai/.cache/zephyr
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Using toolchain: zephyr 0.12.3 (/usr/zephyr-sdk-0.12.3)
-- Found dtc: /usr/zephyr-sdk-0.12.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/boonkhai/zephyrtesting/zephyrproject/zephyr/boards/arm64/qemu_cortex_a53/qemu_cortex_a53.dts
-- Generated zephyr.dts: /home/boonkhai/zephyrtesting/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/boonkhai/zephyrtesting/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/boonkhai/zephyrtesting/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
Parsing /home/boonkhai/zephyrtesting/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/boonkhai/zephyrtesting/zephyrproject/zephyr/boards/arm64/qemu_cortex_a53/qemu_cortex_a53_defconfig'
Merged configuration '/home/boonkhai/zephyrtesting/zephyrproject/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/boonkhai/zephyrtesting/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/boonkhai/zephyrtesting/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
CMake Error at /home/boonkhai/zephyrtesting/zephyrproject/zephyr/cmake/compiler/gcc/target.cmake:10 (message):
  **C compiler /usr/zephyr-sdk-0.12.3//bin/-gcc not found - Please check your
  toolchain installation**
Call Stack (most recent call first):
  /home/boonkhai/zephyrtesting/zephyrproject/zephyr/cmake/target_toolchain.cmake:51 (include)
  /home/boonkhai/zephyrtesting/zephyrproject/zephyr/cmake/app/boilerplate.cmake:592 (include)
  /home/boonkhai/zephyrtesting/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/boonkhai/zephyrtesting/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=/usr/bin/python3 -B/home/boonkhai/zephyrtesting/zephyrproject/zephyr/build -S/home/boonkhai/zephyrtesting/zephyrproject/zephyr/samples/hello_world -GNinja -DBOARD=qemu_cortex_a53


**Environment (please complete the following information):**
 - OS: (Ubuntu 20)
 - Toolchain Zephyr SDK
 - mainline (c4b35c1d36a77f71ecee1086118cff38b5813154)

--- a/cmake/verify-toolchain.cmake
+++ b/cmake/verify-toolchain.cmake
@@ -16,7 +16,11 @@
 # FORMAT=json: Print the output as a json formatted string, useful for Python
 
 # This is the minimum required Zephyr-SDK version which supports CMake package
-set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.11.3)
+if("${ARCH}" STREQUAL "arm64")
+  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12.4)
+else()
+  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12)
+endif()
 
 # Set internal variables if set in environment.
 if(NOT DEFINED ZEPHYR_TOOLCHAIN_VARIANT)
@@ -67,12 +71,12 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
     # To support Zephyr SDK tools (DTC, and other tools) with 3rd party toolchains
     # then we keep track of current toolchain variant.
     set(ZEPHYR_CURRENT_TOOLCHAIN_VARIANT ${ZEPHYR_TOOLCHAIN_VARIANT})
-    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} QUIET HINTS $ENV{ZEPHYR_SDK_INSTALL_DIR})
+    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} REQUIRED QUIET HINTS $ENV{ZEPHYR_SDK_INSTALL_DIR})
     if(DEFINED ZEPHYR_CURRENT_TOOLCHAIN_VARIANT)
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} QUIET PATHS
+    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} REQUIRED QUIET PATHS
                  /usr
 		 /usr/local
 		 /opt

